src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField = false,
+) {
+  const visibleGoods = [...goods].map((good, index) => (
+    {
+      name: good,
+      id: (index + 1),
+      long: good.length,
+    }
+  ));
+
+  if (sortField) {
+    visibleGoods.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return a.name.localeCompare(b.name);
+
+        case SortType.Length:
+          return a.long - b.long;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.None);
+  const [reverseField, setReverseField] = useState(false);
+
+  const goods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortType.Alphabet)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
```

Comment:
If we use the classnames library, it is necessary to divide the classes into separate parts for better maintainability.
```suggestion
          className={cn('button', 'is-info', {
            'is-light': sortField !== SortType.Alphabet,
          })}
```
Fix this in all code.